### PR TITLE
Support Discovery Server / Support Config Server

### PR DESCRIPTION
**Pull request by Mwea**
Added support to config server and discovery server ( and by the way refreshscope ), users can now set their modules.
Todo : Add a bootstrap yml file for each modules..

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,10 @@
 		<spring-integration.version>4.2.4.RELEASE</spring-integration.version>
 		<spring-cloud-stream.version>1.0.0.BUILD-SNAPSHOT</spring-cloud-stream.version>
 		<spring-tuple.version>1.0.0.RC1</spring-tuple.version>
-		<cloudfoundry-client-lib.version>1.1.3</cloudfoundry-client-lib.version>
+        <spring-cloud-context.version>1.0.5.RELEASE</spring-cloud-context.version>
```
Comment (ericbottard) [inline pom.xml]: Please fix indentation (tabs vs spaces)

**Comment by Mwea** [comment]
Tabs fixed

**Comment by markpollack** [comment]
Hi,
I am afraid we can't add @EnableEurekaClient to all of the apps since it is not a requirement of our 'out of the box' apps to have Eureka installed.  If Eureka is not installed, lots of exception traces are shown all over the place. What if you were using Consul for service discovery?  

We have been quite actively thinking of this use case and are working on a version of Spring Initialzr (what powers start.spring.io) that will make it super easy for you to create your own variations of the out of the box applications that add any necessary dependencies and config files to get started.

The new home for 'spring-cloud-stream-modules' will be[ spring-cloud-stream-app-starters](https://github.com/spring-cloud/spring-cloud-stream-app-starters) .  As you can see from the name, the main focus is to help give users a starting point to customize our out of the box apps to their needs.  

The fork of initialzr is just underway (will be incorporated into the main project eventually)[ initialzr-scs](https://github.com/ghillert/initializr-scs) and it up and running here http://start-scs.cfapps.io/ as a means for to test it out, though you can run it from the IDE as well.

Also a maven plugin for spring-cloud-stream-app-starters to generate the directories such as [log-sink-kafka](https://github.com/spring-cloud/spring-cloud-stream-app-starters/tree/master/log/log-sink-rabbit) is just getting started [here](https://github.com/spring-cloud/spring-cloud-stream-app-maven-plugin).

Providing feedback into any of those projects would be extremely welcome!

**Comment by Mwea** [comment]
I'm so happy to see such tools and hope I could use it quickly ! Is there any way to help you to develop other app starters ? In my use case I would be happy to help you to develop a rabbit source and a gpfdist sink because it's what I need but I can do other modules too

**Pull request closed without merging**